oak/proto/grpc_encap.proto

```diff
@@ -35,7 +35,10 @@ message GrpcRequest {
 
 message GrpcResponse {
   int32 stream_id = 1;
-  bytes rsp_msg = 2;
-  google.rpc.Status status = 3;
-  bool last = 4;
+  // has_rsp_msg distinguishes between a rsp_msg that holds google.protobuf.Empty (true)
+  // and the final message that closes a server stream (false).
+  bool has_rsp_msg = 2;
```

Comment:
Alternatively, consider using a nested `Any` message (and similarly for the request).

Reply:
Not sure I follow &ndash; how would that distinguish between an empty message and no-message-present?

Reply:
IIRC nested messages always a have presence bit in the parent message. So you can always do `msg.has_rsp_msg()` and `msg.get_rsp_msg()` which would return the `Any` message (https://developers.google.com/protocol-buffers/docs/proto3#any), which then may or may not have any bytes. Does it make sense?

Reply:
Ah, I thought presence bits were a proto2 feature that went away with proto3, but [reading more](https://cloud.google.com/apis/design/proto3) it seems like they only got dropped for primitive types.   

So as `Any` is a `message` not a primitive, there should still be a `has_rsp_msg()` method (and presumably any `message Envelope { bytes contents=1;}` equivalent would work here &ndash; we're wouldn't be using anything special about `Any` per se).

Thanks for clarifying, I'll take a look tomorrow&hellip;

Reply:
yes, a simple message with a single bytes field would do the job, but I think `Any` would be more semantically accurate. Some languages may offer methods for serializing messages to  and from Any in a more idiomatic way.

Reply:
Added use of `Any`, PTAL.